src/protocol/sort/compose.rs

```diff
@@ -41,27 +34,27 @@ impl<'a, F: Field> Compose<'a, F> {
     /// 4. Revealed permutation is applied locally on another permutation shares (rho)
     /// 5. Unshuffle the permutation with the same random permutations used in step 2, to undo the effect of the shuffling
     #[allow(dead_code)]
-    pub async fn execute(
-        &mut self,
+    pub async fn execute<F: Field>(
         ctx: ProtocolContext<'_, Replicated<F>, F>,
-    ) -> Result<(), BoxError> {
-        let mut random_permutations =
-            get_two_of_three_random_permutations(self.rho.len(), &ctx.prss());
-        let mut random_permutations_copy = random_permutations.clone();
+        sigma: Vec<Replicated<F>>,
+        rho: Vec<Replicated<F>>,
+    ) -> Result<Vec<Replicated<F>>, BoxError> {
+        let random_permutations = get_two_of_three_random_permutations(rho.len(), &ctx.prss());
 
-        Shuffle::new(self.sigma)
-            .execute(ctx.narrow(&ShuffleSigma), &mut random_permutations)
+        let shuffled_sigma = Shuffle::new(sigma, random_permutations.clone())
+            .execute(ctx.narrow(&ShuffleSigma))
             .await?;
 
-        let mut perms = reveal_permutation(ctx.narrow(&RevealPermutation), self.sigma).await?;
+        let revealed_permutation =
+            reveal_permutation(ctx.narrow(&RevealPermutation), &shuffled_sigma).await?;
+        let mut applied_rho = rho;
```

Comment:
What does this do?

Reply:
it converts a immutable variable to a mutable one